From 20803ae92c846590640c7213b43138022bfe912f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Diego=20Elio=20'Flameeyes'=20Petten=C3=B2?= <flameeyes@gmail.com>
Date: Mon, 19 Apr 2010 21:49:17 +0200
Subject: [PATCH] Search within the Ruby library path before the gems.

Older RubyGems versions as shipped with Ruby 1.9 will not search within
paths provided as library paths (RUBYLIB environment variable or -I
option), so will be unable to find locally-present libraries.

To avoid bad interactions with those versions, explicitly search the
library path first, and only then run the command through Gem.
---
 lib/little-plugger.rb |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/lib/little-plugger.rb b/lib/little-plugger.rb
index 17e1be3..fc95adb 100644
--- a/lib/little-plugger.rb
+++ b/lib/little-plugger.rb
@@ -188,6 +188,13 @@ module LittlePlugger
       @loaded ||= {}
       found = {}
 
+      $:.each do |libdir|
+        Dir.glob(File.join(libdir, plugin_path, '*.rb')).each do |path|
+          name = File.basename(path, '.rb').to_sym
+          found[name] = path unless found.key? name
+        end
+      end
+
       Gem.find_files(File.join(plugin_path, '*.rb')).each do |path|
         name = File.basename(path, '.rb').to_sym
         found[name] = path unless found.key? name
-- 
1.7.0.4

